pages/import.tsx

```diff
@@ -58,16 +58,25 @@ const processTrades = async (
     const newSavedData: Partial<ISavedData> = {};
     switch (importDetails.type) {
         case ImportType.TRADES: {
-            const tradesWithFiatRate: ITradeWithFiatRate[] = await addFiatRateToTrades(
-                dataToSave,
-                savedData.settings.fiatCurrency,
-                savedData.settings.fiatRateMethod,
-            );
-            const newTrades: ITradeWithFiatRate[] = sortTrades(
-                savedData.trades.concat(tradesWithFiatRate),
-            ) as ITradeWithFiatRate[];
-            newSavedData.trades = newTrades;
-            break;
+            try {
+                const tradesWithFiatRate: ITradeWithFiatRate[] = await addFiatRateToTrades(
+                    dataToSave,
+                    savedData.settings.fiatCurrency,
+                    savedData.settings.fiatRateMethod,
+                );
+                const newTrades: ITradeWithFiatRate[] = sortTrades(
+                    savedData.trades.concat(tradesWithFiatRate),
+                ) as ITradeWithFiatRate[];
+                newSavedData.trades = newTrades;
+                break;
+            } catch (ex) {
+                setProcessing(false);
+                setAlertData({
```

Comment:
@starsoccer Here
It could be a alert() if you prefer that pop-up dialog style

Reply:
No thats fine